shell/src/main/java/alluxio/cli/fs/command/PersistCommand.java

```diff
@@ -55,63 +85,103 @@ public void validateArgs(CommandLine cl) throws InvalidArgumentException {
   @Override
   protected void runPlainPath(AlluxioURI plainPath, CommandLine cl)
       throws AlluxioException, IOException {
-    persist(plainPath);
+    URIStatus status = mFileSystem.getStatus(plainPath);
+    queuePersist(status);
+  }
+
+  private void queuePersist(URIStatus status) throws AlluxioException, IOException {
+    AlluxioURI uri = new AlluxioURI(status.getPath());
+    if (status.isFolder()) {
+      List<URIStatus> statuses = mFileSystem.listStatus(uri);
+      for (URIStatus s : statuses) {
+        queuePersist(s);
+      }
+    } else if (status.isPersisted()) {
+      System.out.println(status.getPath() + " is already persisted, skipping.");
```

Comment:
Should we omit this for recursively discovered subpaths? This could get noisy if theres a large already-persisted subdirectory

Reply:
Good point, what do you think about logging the total number of skipped paths and then writing the detailed skip info to the `user.log`?

Reply:
I don't see the use case for reporting every individual path that was already persisted. Don't feel strongly about reporting the total number of skipped paths, but would err against it since it suggests something might be wrong when nothing is wrong.

Reply:
Good point, I've removed this.